services/messaging/src/test/java/org/eclipse/hono/messaging/ForwardingDownstreamAdapterTest.java

```diff
@@ -276,35 +276,23 @@ public void testDownstreamLinkClosesUpstreamReceivers() {
         final UpstreamReceiver client = newClient();
         final ProtonSender downstreamSender = newMockSender(false);
 
-        ArgumentCaptor<Handler<AsyncResult<ProtonSender>>> closeHandlerCaptor = ArgumentCaptor.forClass((Class) Handler.class);
-
-        when(downstreamSender.close()).then(invocation -> {
-            closeHandlerCaptor.getValue().handle(Future.succeededFuture(downstreamSender));
-            return downstreamSender;
-        });
-        when(downstreamSender.closeHandler(closeHandlerCaptor.capture())).then(invocation -> {
-            return downstreamSender;
-        });
-
         HandlerCapturingConnectionFactory factory = new HandlerCapturingConnectionFactory(con);
         when(downstreamSender.isOpen()).thenReturn(Boolean.FALSE);
+
         // GIVEN an adapter connected to a downstream container
         givenADownstreamAdapter(downstreamSender);
         adapter.setDownstreamConnectionFactory(factory);
         adapter.start(Future.future());
-        adapter.addSender(client, downstreamSender);
 
         // WHEN the downstream sender is closed
-        adapter.onClientAttach(client, s -> {
-        });
-        downstreamSender.close();
+        adapter.onClientAttach(client, s -> {});
+        ArgumentCaptor<Handler> detachHandlerCaptor = ArgumentCaptor.forClass(Handler.class);
+        verify(downstreamSender).detachHandler(detachHandlerCaptor.capture());
```

Comment:
the three lines above should also go into the `GIVEN` section since they are part of the fixture